### ISSUE TITLE
Chunk render performance issue

### ISSUE THREAD

**Issue by Notlucky101** [RS485/LogisticsPipes#1316]
I notice where I place a pipe next to my wall it connects to it and tries to sink. This causes lag...

**Comment by hron84** [comment]
@Notlucky101 could you please append your Minecraft, Forge and Logistics Pipes version numbers to your report? Also, if you could provide a screenshot about the environment you experience the issue, developers could take a look into it more easily.

**Comment by Notlucky101** [comment]
Ok ok 
Forge  14.23.2838
Lp logisticspipes-0.10.2.183 now its 184, but its been doing it since the first version.

![2019-07-02_08 38 52](https://user-images.githubusercontent.com/42447191/60526412-2f620280-9ca5-11e9-83fc-8ca4107af1fe.png)
![2019-07-02_08 39 22](https://user-images.githubusercontent.com/42447191/60526462-40ab0f00-9ca5-11e9-9c43-09b26bb6ed8b.png)
![2019-07-02_08 40 34](https://user-images.githubusercontent.com/42447191/60526464-40ab0f00-9ca5-11e9-8451-e5d300ac8e4a.png)

**Comment by bziemons** [comment]
They aren't connected to the wall in terms of logic. That's just a visual feature

**Comment by Notlucky101** [comment]
but every time I put a block down on for the wall the lp connects and then lags for a short time.

**Comment by hron84** [comment]
@Notlucky101 
Those pipes do not connect to the wall blocks as they connect to a chest, not even the visuals are the same. They just grow a  "support leg" against the wall just like how real-world pipes are fixed on the wall to prevent bending because of the gravity. If you connect it to a chest, you will find the whole pipe extends against the chest, including those cute red dots too.

The lag you experience is the pipe network recalculation (I guess), it does it every time when you connect a Basic Logistics pipe to discover and register new routes into the network. This is why unrouted pipes exist: if you just want to extend your network in a distance without connecting to an inventory or machine, use unrouted pipes that do exactly what their name says: transports items without any intelligence. They do solve corners but don't do anything in cross-sections. Every time you run into another unrouted pipe and want to cross it, place down a basic logistics pipe.

If you do in this way, you can reduce the network lag to the bare minimum. Also, tend to have smaller networks connected with satellite pipes that help to reduce "pipe adding" lag more.

**Comment by bziemons** [comment]
> The lag you experience is the pipe network recalculation

I don't think so. The code is optimized enough that a pipe without a TileEntity next to it basically runs almost no code, when placed. This is most probably rendering lag that was introduced with 1.12 and it wasn't that bad in 1.7.10. That is because the whole chunk is re-built and we aren't good at that yet. Found a similar performance issue a week ago.

PS: But it is definitely not a problem that those visual wall supports are there ;)

**Comment by hron84** [comment]
@theZorro266 
I've a very good excuse for this: I didn't played with LP on 1.12 and I didn't reached the correspondent stage on Omnifactory yet. So, I didn't know it's rendering lag. On 1.7.10 (where I played with LP a lot) I had that network lag because I had a very poor box to play.

**Comment by Rosethorns** [comment]
I can absolutely say that the rendering lag issue got noticeably worse after Logisitcs Pipes version 10.2.183
